kick_fifo.asm

```diff
@@ -2,49 +2,57 @@
 
 bits 32
 
+%define CACHE_PUSH_MASTER_STATE     0xFD003200
+%define CACHE_PUSH_STATE            0xFD003220
+%define DMA_PUSH_ADDR               0xFD003240
+%define DMA_PULL_ADDR               0xFD003244
+
 kick_fifo:
 
-; Load expected_put into reg
-mov edx, dword [esp+4]
+; Load `expected_push` into EDX
+mov edx, dword [ebp+4]
```

Comment:
I believe this should be `esp+4`. Probably an artifact of a previous version of this file which had `mov ebp, esp` above. Seems to have resolved the issue I posted about for me.

Reply:
Good catch, it is indeed a regression related to the partial revert. Fixed!